File: 02_recursion_basics/02_01_print_name_n_times/PrintNameNTimes.java
public class PrintNameNTimes {
    public static void main(String[] args) {
        int num = 5;
        printNameNTimes(1, num);
    }

    private static void printNameNTimes(int i, int n) {
        if (i > n) {
            return;
        }

        System.out.println("Name");
        printNameNTimes(i + 1, n);
    }
}

File: 02_recursion_basics/02_01_print_name_n_times/print_name_n_times.cpp
#include<bits/stdc++.h>

using namespace std;

void print_n_times(int i, int n) {
    if (i > n) {
        return;
    }

    cout << "Name" << endl;
    print_n_times(i + 1, n);
}

int main() {
    int num = 5;
    print_n_times(1, num);
}